util/http/retry.go

```diff
@@ -89,3 +92,40 @@ func RetryStrategy(ctx context.Context, resp *http.Response, err error) (bool, e
 	}
 	return retry, err2
 }
+
+// ShouldReauthenticate takes an HTTP response and determines whether or not
+// it warrants re-authentication.
+func ShouldReauthenticate(resp *http.Response) bool {
+	defer resp.Body.Close()
```

Comment:
Hmm, we shouldn't consume the response body unless it actually is a 403. I think right now, this function eats the response whether or not there's actually an error, right?

We should also document that this will consume the response body on 403. In the future we might try to log the response body on failure or something which would not work.

Reply:
Oh right, we should close the body if the response warrants re-authentication.  I'll remove this from here and leave it up to the caller to decide what to do with the response.

Reply:
In terms of re-reading the response, we can copy the body to buffer and re-assign in case the caller wants read from it.